client/allocrunner/taskrunner/task_runner.go

```diff
@@ -78,12 +78,18 @@ type TaskRunner struct {
 	// stateDB is for persisting localState and taskState
 	stateDB cstate.StateDB
 
-	// ctx is the task runner's context representing the tasks's lifecycle.
-	// Canceling the context will cause the task to be destroyed.
+	// killCtx is the task runner's context representing the tasks's lifecycle.
+	// The context is canceled when the task is killed.
+	killCtx context.Context
```

Comment:
This can only ever work on the first kill

Reply:
As we discussed out of band `killCtx` is used for _permanently_ killing a task (aka _transitioning it to a terminal state_), so a one-shot context is desired and intended.